From 0c463a00babcc9843c5ce452bd2a0c61f6a66b43 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Mon, 17 Feb 2025 23:46:24 -0600
Subject: [PATCH] trivial: tests: Only run fwupdmgr security tests on x86_64

HSI isn't supported on non-x86 currently, and this causes autopkgtest
failures for Debian.

```
 43s Showing security
 43s cmd: fwupdtool -v security
 43s 19:07:31.573 FuDebug              verbose to info (on console 0)
 43s 19:07:31.573 FuEngine             starting fwupd 2.0.6...
 43s Host Security ID (HSI) is not supported
 43s FAIL: fwupd/fwupdtool.test (Child process exited with code 1)
```
---
 data/tests/fwupdmgr.sh  | 10 ++++++++--
 data/tests/fwupdtool.sh |  8 +++++++-
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/data/tests/fwupdmgr.sh b/data/tests/fwupdmgr.sh
index 7a29dc82e..4b019a4c0 100755
--- a/data/tests/fwupdmgr.sh
+++ b/data/tests/fwupdmgr.sh
@@ -268,15 +268,21 @@ echo "Getting approved firmware..."
 fwupdmgr get-approved-firmware
 rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
+UNAME=$(uname -m)
+if [ "${UNAME}" = "x86_64" ] || [ "${UNAME}" = "x86" ]; then
+       EXPECTED=0
+else
+       EXPECTED=1
+fi
 # ---
 echo "Run security tests..."
 fwupdmgr security
-rc=$?; if [ $rc = 1 ]; then error $rc; fi
+rc=$?; if [ $rc != $EXPECTED ]; then error $rc; fi
 
 # ---
 echo "Run security tests (json)..."
 fwupdmgr security --json
-rc=$?; if [ $rc = 1 ]; then error $rc; fi
+rc=$?; if [ $rc != $EXPECTED ]; then error $rc; fi
 
 # success!
 exit 0
diff --git a/data/tests/fwupdtool.sh b/data/tests/fwupdtool.sh
index 8b512f39f..c521b3c6b 100755
--- a/data/tests/fwupdtool.sh
+++ b/data/tests/fwupdtool.sh
@@ -45,10 +45,16 @@ echo "Showing hwids"
 run hwids
 rc=$?; if [ $rc != 0 ]; then error $rc; fi
 
+UNAME=$(uname -m)
+if [ "${UNAME}" = "x86_64" ] || [ "${UNAME}" = "x86" ]; then
+       EXPECTED=0
+else
+       EXPECTED=1
+fi
 # ---
 echo "Showing security"
 run security
-rc=$?; if [ $rc != 0 ]; then error $rc; fi
+rc=$?; if [ $rc != $EXPECTED ]; then error $rc; fi
 
 # ---
 echo "Showing plugins"
-- 
2.43.0

